### PR TITLE
New dask-jobqueue 0.3.0, docrep 0.2.3. Updated distributed 1.15.1 -> 1.22.1

### PR DESCRIPTION
**Pull request by costrouc**
###### motivation

Adding python package `dask-jobqueue`. `dask-jobqueue` has required
dependencies of new package `docrep` and updated version of
`distributed`.

The newest release of `distributed` has a change in
dependencies. `msgpack-python` (depreciated) -> `msgpack` and `pyyaml`
is a new dependency.

###### Things done

Added python package `dask-distributed` 0.3.0 and `docrep` 0.2.3.
Update python package `distributed` 1.15.1 -> 1.22.1

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dotlambda** [comment]
Instead of removing `meta.maintainers`, why not add yourself to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix? This should be done in an additional commit before the existing ones.
Also, please create a separate commit for each package and format the commit messages according to the guidelines, e.g. `pythonPackages.docrep: init at 0.2.3`.

File: pkgs/development/python-modules/dask-jobqueue/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  # tests required slurm, sge, schedulers to be available
+  doCheck = false;
+
+  meta = {
+    description = "Easy deployment of Dask Distributed on job queuing systems such as PBS, Slurm, or SGE.*";
+    homepage = https://github.com/dask/dask-jobqueue;
+    license = lib.licenses.bsd3;
+  };
+}
```
Comment (dotlambda) [inline pkgs/development/python-modules/dask-jobqueue/default.nix]: final \n

File: pkgs/development/python-modules/docrep/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  # tests not packaged with PyPi download
+  doCheck = false;
+
+  meta = {
+    description = "Python package for docstring repetition";
+    homepage = https://github.com/Chilipp/docrep;
+    license = lib.licenses.gpl2;
+  };
+}
```
Comment (dotlambda) [inline pkgs/development/python-modules/docrep/default.nix]: final \n

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -1979,17 +1983,17 @@ in {
   distributed = buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: This expression needs to be moved to `pkgs/development/python-modules/distributed/default.nix`.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -1979,17 +1983,17 @@ in {
...
 
     name = "distributed-${version}";
-    version = "1.15.1";
+    version = "1.22.1";
 
     src = pkgs.fetchurl {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: `fetchPypi`

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -1979,17 +1983,17 @@ in {
   distributed = buildPythonPackage rec {
 
     name = "distributed-${version}";
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please specify `pname = "distributed"` instead.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -1979,17 +1983,17 @@ in {
...
-    version = "1.15.1";
+    version = "1.22.1";
 
     src = pkgs.fetchurl {
       url = "mirror://pypi/d/distributed/${name}.tar.gz";
-      sha256 = "037a07sdf2ch1d360nqwqz3b4ld8msydng7mw4i5s902v7xr05l6";
+      sha256 = "eefdd511912a001077bf1e00e2c3989dcfa853d68d14a8cd5b688c6175389e3a";
     };
 
     buildInputs = with self; [ pytest docutils ];
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: `checkInputs`

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by costrouc** [comment]
Closing and separating into separate pull requests.

**Comment by FRidh** [comment]
@costrouc don't, just create separate commits in a single PR.

**Comment by costrouc** [comment]
@FRidh I have already done this sorry. Definitely will just do separate commits in the future because that is much much easier.

See these:

 - #44665
 - #44657
- #44654
- #44653

**Pull request closed without merging**